napari/utils/_tests/test_proxies.py

```diff
@@ -115,3 +137,31 @@ def test_receive_return_proxy_object():
 def test_viewer_method():
     viewer = PublicOnlyProxy(ViewerModel())
     assert viewer.add_points() is not None
+
+
+def test_proxy_fixture_warning(make_napari_viewer_proxy, monkeypatch):
+    viewer = make_napari_viewer_proxy()
+
+    monkeypatch.setattr(misc, 'ROOT_DIR', '/some/other/package')
+    with pytest.warns(FutureWarning, match='Private attribute access'):
+        viewer.window._qt_window
+
+
+def test_proxy_fixture_threat_warning(make_napari_viewer_proxy):
+    viewer = make_napari_viewer_proxy()
+
+    ok = False
+
+    class MyThread(Thread):
+        def run(self):
+            nonlocal ok
+
+            with pytest.raises(RuntimeError, match='Setting attributes'):
+                viewer.status = 'hi'
+
+            ok = True
+
+    t = MyThread()
+    t.start()
+    t.join()
+    assert ok
```

Comment:
Optional: As I read it, this looks like `test_thread_proxy_guard` but with the napari viewer. If that's the case, I'd consider matching the style of those with executors rather than using `Thread`:

```suggestion
def test_proxy_fixture_thread_error(make_napari_viewer_proxy, single_threaded_executor):
    viewer = make_napari_viewer_proxy()
    future = single_threaded_executor.submit(viewer.__setattr__, 'status', 'hi')
    with pytest.raises(RuntimeError, match='Setting attributes'):
        future.result()
```

For me, that makes it clearer what this is testing as the `with pytest.raises` statement doesn't get so buried. Whereas in the original form `assert ok` seems like the main thing we're testing, whereas that's mostly checking that `with pytest.raises` swallows the right exception.

I guess the downsides of using the executor are that it doesn't actually raise the exception until you call `result`  and the ugliness of `__setattr__`. I'll leave it up to you.

Reply:
Whatever you choose, fix the name of the test though - it currently ends in `threat_warning`, whereas I think it should be something like `thread_error`.

Reply:
Your implementation is shorter so I prefer it.